tests/test_swin_unetr.py

```diff
@@ -0,0 +1,86 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from unittest import skipUnless
+
+import torch
+from parameterized import parameterized
+
+from monai.networks import eval_mode
+from monai.networks.nets.swin_unetr import SwinUNETR
+from monai.utils import optional_import
+
+einops, has_einops = optional_import("einops")
+
+TEST_CASE_UNETR = []
+for attn_drop_rate in [0.4]:
+    for in_channels in [1]:
+        for depth in [[2, 1, 1, 1], [1, 2, 1, 1]]:
+            for out_channels in [2]:
+                for img_size in [64]:
+                    for feature_size in [12]:
+                        for norm_name in ["instance"]:
+                            test_case = [
+                                {
+                                    "in_channels": in_channels,
+                                    "out_channels": out_channels,
+                                    "img_size": (img_size,) * 3,
+                                    "feature_size": feature_size,
+                                    "depths": depth,
+                                    "norm_name": norm_name,
+                                    "attn_drop_rate": attn_drop_rate,
+                                },
+                                (2, in_channels, *([img_size] * 3)),
+                                (2, out_channels, *([img_size] * 3)),
+                            ]
+                            TEST_CASE_UNETR.append(test_case)
+
+
+class TestPatchEmbeddingBlock(unittest.TestCase):
+    @parameterized.expand(TEST_CASE_UNETR)
+    @skipUnless(has_einops, "Requires einops")
+    def test_shape(self, input_param, input_shape, expected_shape):
+        net = SwinUNETR(**input_param)
+        with eval_mode(net):
+            result = net(torch.randn(input_shape))
+            self.assertEqual(result.shape, expected_shape)
+
+    def test_ill_arg(self):
+        with self.assertRaises(ValueError):
+            SwinUNETR(
+                in_channels=1,
+                out_channels=3,
+                img_size=(128, 128, 128),
+                feature_size=24,
+                norm_name="instance",
+                attn_drop_rate=4,
+            )
+
+        with self.assertRaises(ValueError):
+            SwinUNETR(in_channels=1, out_channels=2, img_size=(96, 96), feature_size=48, norm_name="instance")
+
+        with self.assertRaises(ValueError):
+            SwinUNETR(in_channels=1, out_channels=4, img_size=(96, 96, 96), feature_size=50, norm_name="instance")
+
+        with self.assertRaises(ValueError):
```

Comment:
would be great to test the compatibility with torchscript

Reply:
That's a good point, thanks for raising it!

Reply:
We do not support torch script in this implementation.

Reply:
According to [MONAI contribution guide](https://github.com/Project-MONAI/MONAI/blob/dev/CONTRIBUTING.md), support of torch script is not requirement, hence shall not be a blocker.

Reply:
Hi @ahatamiz ,

If it's possible to support TorchScript somehow, I would suggest to support it, especially for so hot a model.
As you can see, we got much feedback from Clara users, SA team guys, and customers about applying our state-of-the-art model in their inference pipeline. TorchScript is very useful in real deploy cases.
And we also discussed the future plan to optimize `swin unetr` pipeline soon for better performance and involve more end users.
What do you think?

Thanks in advance.

Reply:
Hi @Nic-Ma 

My concern is that there is no specific development requirements that is consistently enforced for different models. Even apart from this model, a developer need to know if:

1. Support of both 2D and 3D modalities is a must ?  Some networks are inherently 3D or 2D and their extension, although possible, may not be optimal. 
2. Support of Torchscript is required, even if some operations needs to be changed to realize it, despite deviating from reported performance ? 

For mmar, we already have a solution for Torchscript conversion for models that may not support it. 

Thanks,